MapifyEditor/Tools/TrackToolsWindow.cs

```diff
@@ -0,0 +1,1908 @@
+using Mapify.Editor.Utils;
+using UnityEditor;
+using UnityEngine;
+using static Mapify.Editor.Tools.ToolEnums;
+
+#if UNITY_EDITOR
+namespace Mapify.Editor.Tools
+{
+    public class TrackToolsWindow : EditorWindow
+    {
+        [MenuItem("Mapify/Track Tools")]
+        public static void ShowWindow()
+        {
+            var window = GetWindow<TrackToolsWindow>();
+            window.Show();
+            window.titleContent = new GUIContent("Track Tools");
+            window.autoRepaintOnSceneChange = true;
+        }
+
+        #region FIELDS
+
+        // Selection.
+        private bool _showSelection = true;
+        private SelectionType _selectionType = SelectionType.None;
+        private Track[] _selectedTracks = new Track[0];
+        private BezierPoint[] _selectedPoints = new BezierPoint[0];
+        private bool _showTracks = true;
+        private bool _showPoints = true;
+
+        // Track creation.
+        private bool _showCreation = true;
+        private CreationMode _currentMode = CreationMode.Straight;
+        private Transform _currentParent;
+        private TrackOrientation _orientation;
+        private float _endGrade = 0.0f;
+        private float _trackDistance = 4.5f;
+
+        // Straight.
+        private float _length = 100.0f;
+
+        // Curves.
+        private float _radius = 500.0f;
+        private float _arc = 45.0f;
+        private float _maxArcPerPoint = 22.5f;
+        private bool _changeArc = false;
+
+        // Switches.
+        private SwitchPoint _connectingPoint = SwitchPoint.Joint;
+
+        // Yards.
+        private YardOptions _yardOptions = YardOptions.DefaultOptions;
+        private bool _showYardCache = false;
+
+        // Turntable.
+        private TurntableOptions _turntableOptions = TurntableOptions.DefaultOptions;
+
+        // Special.
+        private SpecialTrack _currentSpecial = SpecialTrack.Buffer;
+        private bool _useHandle2Start = false;
+        private bool _useHandle2End = false;
+        private float _lengthMultiplier = 1.0f;
+        private bool _isTrailing = false;
+        private float _switchDistance = 6.0f;
+        private float _crossAngle = 20.0f;
+
+        // Track editing.
+        private bool _showEditing = false;
+        private EditingMode _editingMode = EditingMode.Merge;
+
+        // Track prefabs.
+        private bool _showPrefabs = false;
+        public Track TrackPrefab;
+        public BufferStop BufferPrefab;
+        public Switch LeftSwitch;
+        public Switch RightSwitch;
+        public Turntable TurntablePrefab;
+
+        // Settings.
+        private bool _showSettings = false;
+        private Color _forwardColour = Color.cyan;
+        private Color _backwardColour = Color.red;
+        private Color _newColour = Color.green;
+        private int _sampleCount = 8;
+
+        // Contents for the tool selection.
+        private readonly GUIContent[] _modeContents = new GUIContent[]
+        {
```

Comment:
Arrays created like this don't need an explicit type
```suggestion
        private readonly GUIContent[] _modeContents = {